playbooks/common/openshift-cluster/upgrades/v3_9/label_nodes.yml

```diff
@@ -0,0 +1,48 @@
+---
+
+- hosts: oo_first_master
+  roles:
+    - openshift_facts
+    - lib_openshift
+  post_tasks:
+    - when: osm_default_node_selector is not defined
+      block:
+        - name: Retrieve non-infra, non-master nodes
+          oc_obj:
+            state: list
+            kind: Node
+            selector: '{{ (openshift_hosted_infra_selector | default("region=infra")) | regex_replace("=", "!=") }},node-role.kubernetes.io/master!=true'
+          register: result
+
+        - set_fact:
+            non_master_non_infra_nodes: "{{ result.results.results.0['items'] | map(attribute='metadata') | map(attribute='name') | list }}"
+
+        - name: label non-master non-infra nodes compute
+          oc_label:
+            name: '{{ item }}'
+            kind: node
+            state: add
+            labels:
+              - key: node-role.kubernetes.io/compute
+                value: true
+          with_items: '{{ non_master_non_infra_nodes }}'
+
+- hosts: oo_masters_to_config
+  roles:
+    - openshift_facts
+  post_tasks:
+    - when: osm_default_node_selector is not defined
```

Comment:
These tasks belong in the openshift_master role; let's use import_role + tasks_from for this.

Reply:
I think the lineinfile task that changes the default node selector should probably stay in the playbook, it's really only relevant to this specific upgrade. The restart is already implemented in openshift_master though, so that should definitely be imported, and I can update the default defaultNodeSelector in openshift_master as well